src/Http/GraphRequest.php

```diff
@@ -139,21 +124,27 @@ public function getReturnsStream()
     */
     public function setAccessToken(string $accessToken): self
     {
+        unset($this->headers['Authorization']); // Prevents appending new token
         $this->addHeaders(['Authorization' => 'Bearer '.$accessToken]);
         return $this;
     }
 
     /**
-    * Sets the return type of the response object
-    *
-    * @param string $returnClass The class name to use
-    *
-    * @return $this object
-    */
+     * Sets the return type of the response object
+     * Can be set to a model or \Psr\Http\Message\StreamInterface
+     *
+     * @param string $returnClass The class name to use
+     *
+     * @return $this object
+     * @throws GraphClientException when $returnClass is not an existing class
+     */
     public function setReturnType(string $returnClass): self
     {
+        if (!class_exists($returnClass) && !interface_exists($returnClass)) {
+            throw new GraphClientException("Return type specified does not match an existing class definition");
+        }
         $this->returnType = $returnClass;
-        if ($this->returnType == "GuzzleHttp\Psr7\Stream") {
+        if ($this->returnType == StreamInterface::class) {
```

Comment:
```suggestion
        if ($this->returnType === StreamInterface::class) {
```

Reply:
Updated. Although we're guaranteed of a string $returnClass due to the strong argument type